pkg/nodeprofilemanager/v1/nodeprofile_manager.go

```diff
@@ -0,0 +1,216 @@
+package nodeprofilemanager
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"node-agent/pkg/config"
+	"node-agent/pkg/nodeprofilemanager"
+	"node-agent/pkg/objectcache"
+	"node-agent/pkg/relevancymanager"
+	"node-agent/pkg/rulemanager"
+	"node-agent/pkg/utils"
+	"time"
+
+	"github.com/armosec/armoapi-go/armotypes"
+	"github.com/armosec/utils-k8s-go/armometadata"
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+	"golang.org/x/net/context"
+	v1 "k8s.io/api/core/v1"
+)
+
+type NodeProfileManager struct {
+	clusterData      armometadata.ClusterConfig
+	config           config.Config
+	httpClient       *http.Client
+	k8sObjectCache   objectcache.K8sObjectCache
+	nodeName         string
+	relevancyManager relevancymanager.RelevancyManagerClient
+	ruleManager      rulemanager.RuleManagerClient
+}
+
+func NewNodeProfileManager(config config.Config, clusterData armometadata.ClusterConfig, nodeName string, k8sObjectCache objectcache.K8sObjectCache, relevancyManager relevancymanager.RelevancyManagerClient, ruleManager rulemanager.RuleManagerClient) *NodeProfileManager {
+	return &NodeProfileManager{
+		clusterData: clusterData,
+		config:      config,
+		httpClient: &http.Client{
+			Timeout: time.Duration(config.Exporters.HTTPExporterConfig.TimeoutSeconds) * time.Second,
+		},
+		k8sObjectCache:   k8sObjectCache,
+		nodeName:         nodeName,
+		relevancyManager: relevancyManager,
+		ruleManager:      ruleManager,
+	}
+}
+
+var _ nodeprofilemanager.NodeProfileManagerClient = (*NodeProfileManager)(nil)
+
+func (n *NodeProfileManager) Start(ctx context.Context) {
+	go func() {
+		time.Sleep(n.config.InitialDelay)
+		for {
+			time.Sleep(n.config.NodeProfileInterval)
+			profile, err := n.getProfile()
+			if err != nil {
+				logger.L().Ctx(ctx).Warning("get profile error", helpers.Error(err))
+			} else {
+				err := n.sendProfile(profile)
+				if err != nil {
+					logger.L().Ctx(ctx).Warning("send profile error", helpers.Error(err))
+				}
+			}
+		}
+	}()
+}
+
+func (n *NodeProfileManager) getProfile() (*armotypes.NodeProfile, error) {
+	profile := &armotypes.NodeProfile{
+		PodStatuses:             []armotypes.PodStatus{},
+		CurrentState:            "Running",
+		NodeAgentRunning:        true,
+		RuntimeDetectionEnabled: n.config.EnableRuntimeDetection,
+	}
+	for _, pod := range n.k8sObjectCache.GetPods() {
+		var app string
+		if pod.Labels != nil {
+			app = pod.Labels["app"]
```

Comment:
What if this label is empty? Consider looking at the `app.kubernetes.io/name` label as well